src/hidra/receiver/asapo_transfer.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import logging
+
+from hidra import Transfer, __version__, generate_filepath
+from plugins.asapo_producer import AsapoWorker
+
+logger = logging.getLogger(__name__)
+
+
+class AsapoTransfer:
+    def __init__(self, asapo_worker, signal_host, target_host, target_port, target_dir):
+
+        self.signal_host = signal_host
+        self.targets = [[target_host, target_port, 1]]
+        self.target_dir = target_dir
+        self.asapo_worker = asapo_worker
+        self.query = Transfer("STREAM_METADATA", self.signal_host)
+        self.query.initiate(self.targets)
+
+    def start_transfer(self):
+        self.query.start()
+        self.run()
+
+    def run(self):
+        while True:
+            try:
+                [metadata, data] = self.query.get()
+                if metadata is not None and data is not None:
+                    local_path = generate_filepath(self.target_dir, metadata)
+                    logger.info("Send file {local_path}".format(local_path=local_path))
+                    self.asapo_worker.send_message(local_path, metadata)
+            except Exception as e:
+                logger.error("Transmission does not succeed. {e}. File ignored".format(e=str(e)))
```

Comment:
You need to recover somehow in case of errors. I am not sure that simply retrying is enough, it might be that in some circumstances you need to reconnect.

Reply:
How to check, if connection is still alive?